### ISSUE TITLE
In version 6.10.11: Failed launching test session: TypeError: hooks.map is not a function

### ISSUE THREAD

**Issue by GRme** [webdriverio/webdriverio#6242]
**Environment (please complete the following information):**
 - **WebdriverIO version:** 6.10.11
 - **Mode:** Local Runner
 - **If WDIO Testrunner, running sync/async:** sync
 - **Node.js version:** 10.21.0
 - **NPM version:** 6.14.5
 - **Browser name and version:** Chrome 87.0.4280.88
 - **Platform name and version:** Mac OS X 10.15.7
 - **Additional wdio packages used (if applicable):** 
     "@wdio/appium-service": "^6.4.7",
    "@wdio/cli": "^6.5.2",
    "@wdio/cucumber-framework": "6.0.13",
    "@wdio/devtools-service": "^6.5.2",
    "@wdio/local-runner": "^6.10.11",
    "@wdio/selenium-standalone-service": "^6.4.0",
    "@wdio/spec-reporter": "5.22.4",
    "@wdio/sync": "6.0.12",

**Describe the bug**
Running the tests with local runner causes in a general error. With local-runner version 6.10.10 everything is fine, but the new version causes an error.

I'm starting my tests with `npx wdio wdio-configs/wdio.conf.chrome.js --cucumberOpts.tagExpression=@myTag`

Regarding the error message below I think maybe my hooks could be interesting for you:

```
  onPrepare: () => {
    // Remove the `tmp/` folder that holds the json report files
    removeSync(jsonTmpDirectory);
    if (!fs.existsSync(jsonTmpDirectory)) {
      fs.mkdirSync(jsonTmpDirectory);
    }
  },

  before: function(capabilities, specs) {
    // Add custom commands to WebdriverIO
    Object.keys(commands).forEach(key => {
      browser.addCommand(key, commands[key]);
    });
  },

  onComplete: () => {
    try {
      let consolidatedJsonArray = wdioParallel.getConsolidatedData({
        parallelExecutionReportDirectory: jsonTmpDirectory
      });

      let jsonFile = `${jsonTmpDirectory}report.json`;
      fs.writeFileSync(jsonFile, JSON.stringify(consolidatedJsonArray));

      var options = {
        theme: "bootstrap",
        jsonFile: jsonFile,
        output: `tests/reports/html/report-${currentTime}.html`,
        reportSuiteAsScenarios: true,
        scenarioTimestamp: true,
        launchReport: false,
        ignoreBadJsonFile: true
      };

      reporter.generate(options);
    } catch (err) {
      console.log("err", err);
    }
  }
```

**To Reproduce**

[Include code or an example repository that can easily be set up]

**Expected behavior**

**Log**

```
[0-0] 2020-12-17T13:55:49.531Z ERROR @wdio/local-runner: Failed launching test session: TypeError: hooks.map is not a function
    at Object.executeHooksWithArgs (/Users/GRme/projects/Test/github/Testautomation/node_modules/@wdio/sync/build/executeHooksWithArgs.js:25:17)
    at Runner.run (/Users/GRme/projects/Test/github/Testautomation/node_modules/@wdio/runner/build/index.js:44:23)
    at process._tickCallback (internal/process/next_tick.js:68:7)
/Users/GRme/projects/Test/github/Testautomation/node_modules/@wdio/cli/build/interface.js:127
            throw new Error('Could not find job');
            ^

Error: Could not find job
    at WDIOCLInterface.clearJob (/Users/GRme/projects/Test/github/Testautomation/node_modules/@wdio/cli/build/interface.js:127:19)
    at WDIOCLInterface.emit (events.js:198:13)
    at WDIOCLInterface.EventEmitter.emit (domain.js:448:20)
    at Launcher.endHandler (/Users/GRme/projects/Test/github/estautomation/node_modules/@wdio/cli/build/launcher.js:222:28)
    at WorkerInstance.emit (events.js:198:13)
    at WorkerInstance.EventEmitter.emit (domain.js:448:20)
    at WorkerInstance._handleExit (/Users/GRme/projects/Test/github/Testautomation/node_modules/@wdio/local-runner/build/worker.js:93:14)
    at ChildProcess.emit (events.js:198:13)
    at ChildProcess.EventEmitter.emit (domain.js:448:20)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:248:12)
scripty ERR! script failed: '/Users/GRme/projects/Test/github/Testautomation/scripts/clear-test.sh'
scripty ERR! exit status: 1
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! testautomation@0.0.1 clear-test: `scripty "wdio-configs/wdio.conf.chrome.js --cucumberOpts.tagExpression=@mytag"`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the testautomation@0.0.1 clear-test script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/GRme/.npm/_logs/2020-12-17T13_55_49_585Z-debug.log
``

**Comment by christian-bromann** [comment]
@GRme could you provide a reproducible example? I can't reproduce this? Did you make sure to update all wdio related packages?

**Comment by abkoyuncu** [comment]
I was having the same issue for mocha. 

It was working fine with cucumber and then I switched to mocha and started to have above issue.
I noticed that @dio/cucumber-framework package was 6.5.0 as well as above mentioned packages whereas I had @wdio/mocha-framework in version 6.10.11.


Upgrading to version 6.10.11 helped for below packages:
-@wdio/cli
-@wdio/sync
-@wdio/local-runner

@GRme I suggest upgrading cucumber and other packages to most compatible versions. 
6.5.0 was working for me for cucumber
